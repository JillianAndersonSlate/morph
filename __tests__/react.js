import { morph } from '../index.js'
import { join } from 'path'
import { existsSync, readdirSync, readFileSync } from 'fs'

const isView = f => /\.view$/.test(f)
const getPath = (f = '.') => join(__dirname, 'views', f)
;['react-dom', 'react-native'].forEach(as =>
  describe(as, () => {
    readdirSync(getPath())
      .filter(isView)
      .forEach(f => {
        const name = f.replace(/\.view$/, '')
        const code = readFileSync(getPath(f), 'utf-8')
        const testFile = getPath(`${f}.tests`)
        const tests = existsSync(testFile)
          ? readFileSync(testFile, 'utf-8')
          : undefined

        it(`parses ${as} ${name}`, () => {
          const morphed = morph(code, {
            as,
            inlineStyles: true,
            name,
            pretty: true,
            tests,
          })
          expect(morphed).toMatchSnapshot()
        })
        // TODO test rendered morphed view
      })
  })
)
