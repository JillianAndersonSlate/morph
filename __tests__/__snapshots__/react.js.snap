// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-dom parses react-dom Action 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const Action = props =>
  props.check
    ? <button onClick={props.clickMe} className={\`\${styles.h1lmw638}\`}>
        <div>click me</div>
      </button>
    : null;

export default Action;
"
`;

exports[`react-dom parses react-dom BackgroundImage 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  hzr7mx4: css('css-1bxi9t2', [], () => [
    \`.css-1bxi9t2 { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column;
background-image: url(\\"https://viewsdx.com/image.jpg\\");
background-size: cover; }\`,
  ]),
  hg3iw2q: css('css-sw3yj', [], () => [
    \`.css-sw3yj { flex-direction: row; -webkit-box-direction: normal; -webkit-box-orient: horizontal; -ms-flex-direction: row; -webkit-flex-direction: row; }\`,
  ]),
};

const BackgroundImage = props =>
  <div className={\`\${styles.hzr7mx4}\`}>
    <div
      className={\`\${styles.hg3iw2q}\`}
      style={{
        backgroundImage: \`url(\${props.backgroundImage})\`,
        backgroundSize: cover,
      }}
    />
  </div>;

export default BackgroundImage;
"
`;

exports[`react-dom parses react-dom BoxShadow 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const BoxShadow = props =>
  <div
    className={\`\${styles.h1lmw638}\`}
    style={{ boxShadow: \`10px 5px 2px \${props.color}\` }}
  />;

export default BoxShadow;
"
`;

exports[`react-dom parses react-dom CaptureEmail 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  hevllt7: css('css-17byui2', [], () => [
    \`.css-17byui2 { border: 1px solid black;
color: red; }\`,
    \`.css-17byui2::placeholder { color: blue; }\`,
  ]),
};

const CaptureEmail = props =>
  <input onSubmit={props.onSubmit} className={\`\${styles.hevllt7}\`} />;

export default CaptureEmail;
"
`;

exports[`react-dom parses react-dom CaptureFile 1`] = `
"import React from 'react';

const CaptureFile = props => <input />;

export default CaptureFile;
"
`;

exports[`react-dom parses react-dom CaptureNumber 1`] = `
"import React from 'react';

const CaptureNumber = props => <input />;

export default CaptureNumber;
"
`;

exports[`react-dom parses react-dom CapturePhone 1`] = `
"import React from 'react';

const CapturePhone = props => <input />;

export default CapturePhone;
"
`;

exports[`react-dom parses react-dom CaptureSecure 1`] = `
"import React from 'react';

const CaptureSecure = props => <input />;

export default CaptureSecure;
"
`;

exports[`react-dom parses react-dom CaptureText 1`] = `
"import React from 'react';

const CaptureText = props => <input />;

export default CaptureText;
"
`;

exports[`react-dom parses react-dom GoTo 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const GoTo = props =>
  <a
    target=\\"_blank\\"
    href=\\"https://viewsdx.com\\"
    className={\`\${styles.h1lmw638}\`}
  >
    <div>Go to ViewsDX website!</div>
  </a>;

export default GoTo;
"
`;

exports[`react-dom parses react-dom Horizontal 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1nnugsi: css('css-vkz5lf', [], () => [
    \`.css-vkz5lf { flex-direction: row; -webkit-box-direction: normal; -webkit-box-orient: horizontal; -ms-flex-direction: row; -webkit-flex-direction: row;
border: 1px solid red; }\`,
  ]),
};

const Horizontal = props => <div className={\`\${styles.h1nnugsi}\`} />;

export default Horizontal;
"
`;

exports[`react-dom parses react-dom Hover 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  hpg1h4y: css('css-i58qmr', [], () => [
    \`.css-i58qmr { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column;
background-color: black; }\`,
    \`.css-i58qmr:hover { background-color: white; }\`,
  ]),
  h1669j73: css('css-9glvsj', [], () => [
    \`.css-9glvsj { color: white; }\`,
    \`.css-9glvsj:hover { color: black; }\`,
  ]),
};

const Hover = props =>
  <button onClick={props.onClick} className={\`\${styles.hpg1h4y}\`}>
    <div className={\`\${styles.h1669j73}\`}>Hey!</div>
  </button>;

export default Hover;
"
`;

exports[`react-dom parses react-dom HoverProxy 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  hpg1h4y: css('css-i58qmr', [], () => [
    \`.css-i58qmr { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column;
background-color: black; }\`,
    \`.css-i58qmr:hover { background-color: white; }\`,
  ]),
  h1669j73: css('css-9glvsj', [], () => [
    \`.css-9glvsj { color: white; }\`,
    \`.css-9glvsj:hover { color: black; }\`,
  ]),
};

const HoverProxy = props =>
  <button onClick={props.onClick} className={\`\${styles.hpg1h4y}\`}>
    <props.icon from={props.icon} fill=\\"white\\" fillHover=\\"black\\" />
    <div className={\`\${styles.h1669j73}\`}>Hey!</div>
  </button>;

export default HoverProxy;
"
`;

exports[`react-dom parses react-dom Image 1`] = `
"import React from 'react';

const Image = props => <img source=\\"https://viewsdx.com/rocks.jpg\\" />;

export default Image;
"
`;

exports[`react-dom parses react-dom JustText 1`] = `
"import React from 'react';

const JustText = props => <div>hey!</div>;

export default JustText;
"
`;

exports[`react-dom parses react-dom ListOfSomething 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const ListOfSomething = props =>
  <div className={\`\${styles.h1lmw638}\`}>
    {props.list.map((item, i) => <div key={i}>{item.name}</div>)}
  </div>;

export default ListOfSomething;
"
`;

exports[`react-dom parses react-dom ListWithData 1`] = `
"import React from 'react';
import css from 'glam';
import usersData from './users.data.js';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const ListWithData = props =>
  <div className={\`\${styles.h1lmw638}\`}>
    {usersData.map((item, i) => <div key={i}>{item.name}</div>)}
  </div>;

export default ListWithData;
"
`;

exports[`react-dom parses react-dom NameIsType 1`] = `
"import React from 'react';

const NameIsType = props => <div>iam</div>;

export default NameIsType;
"
`;

exports[`react-dom parses react-dom NestedRoutes 1`] = `
"import React from 'react';
import { Link } from 'react-router-dom';
import { Route } from 'react-router-dom';
import Topic from './Topic.view.js';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const NestedRoutes = props =>
  <div>
    <div>topics</div>
    <Link to=\\"rendering\\" className={\`\${styles.h1lmw638}\`}>
      <div>Rendering with React</div>
    </Link>
    <Route
      path={\`\${props.match.url}/:topic\`}
      render={routeProps => <Topic {...routeProps} />}
    />
  </div>;

export default NestedRoutes;
"
`;

exports[`react-dom parses react-dom Print 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  hrak8un: css('css-kbgujp', [], () => [
    \`@media print {
    .css-kbgujp {
        display: none
    }
}\`,
  ]),
};

const Print = props =>
  <div>
    <div className={\`\${styles.hrak8un}\`}>not in print</div><div>hey</div>
  </div>;

export default Print;
"
`;

exports[`react-dom parses react-dom Proxy 1`] = `
"import React from 'react';

const Proxy = props =>
  <div><props.icon from={props.icon} fill=\\"red\\" /><div>{props.text}</div></div>;

export default Proxy;
"
`;

exports[`react-dom parses react-dom Router 1`] = `
"import React from 'react';
import About from './About.view.js';
import Home from './Home.view.js';
import { Route } from 'react-router-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Topics from './Topics.view.js';

const Router = props =>
  <Router>
    <div>
      <Route
        path=\\"/\\"
        exact
        render={routeProps =>
          <Home heigth={props.height} width={360} {...routeProps} />}
      />
      <Route path=\\"/about\\" render={routeProps => <About {...routeProps} />} />
      <Route path=\\"/topics\\" render={routeProps => <Topics {...routeProps} />} />
    </div>
  </Router>;

export default Router;
"
`;

exports[`react-dom parses react-dom Svg 1`] = `
"import React from 'react';

const Svg = props =>
  <svg viewBox=\\"0 0 20 20\\">
    <path d=\\"0 20\\" /><circle cx={10} cy={5} r={5} />
    <g><path d={1} /><path d={2} /></g><text text=\\"hey\\" />
  </svg>;

export default Svg;
"
`;

exports[`react-dom parses react-dom Teleport 1`] = `
"import React from 'react';
import { Link } from 'react-router-dom';
import css from 'glam';

const styles = {
  hg3iw2q: css('css-sw3yj', [], () => [
    \`.css-sw3yj { flex-direction: row; -webkit-box-direction: normal; -webkit-box-orient: horizontal; -ms-flex-direction: row; -webkit-flex-direction: row; }\`,
  ]),
};

const Teleport = props =>
  <Link to=\\"..\\" className={\`\${styles.hg3iw2q}\`}><div>go back</div></Link>;

export default Teleport;
"
`;

exports[`react-dom parses react-dom Text 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1w7s7t6: css('css-1l09f9n', [], () => [
    \`.css-1l09f9n { color: blue;
font-family: Montserrat, sans-serif;
font-size: 16px;
font-weight: 300; }\`,
  ]),
};

const Text = props =>
  <div className={\`\${styles.h1w7s7t6}\`}>Hi, I'm some text</div>;

export default Text;
"
`;

exports[`react-dom parses react-dom Toggle 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

class RemapToggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isOn: props.isOn };
  }
  toggleIsOn = () => this.setState({ isOn: !this.state.isOn });

  render() {
    return (
      <Toggle {...this.props} {...this.state} toggleIsOn={this.toggleIsOn} />
    );
  }
}

const Toggle = props =>
  <button onClick={props.toggleIsOn} className={\`\${styles.h1lmw638}\`} />;

export default RemapToggle;
"
`;

exports[`react-dom parses react-dom UseOfProxy 1`] = `
"import React from 'react';
import Button from './Button.view.js';
import MyIcon from './MyIcon.view.js';

const UseOfProxy = props =>
  <div><Button color=\\"red\\" icon={MyIcon} text=\\"I'm a button!\\" /></div>;

export default UseOfProxy;
"
`;

exports[`react-dom parses react-dom Vertical 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1vjf5pn: css('css-nxb3y5', [], () => [
    \`.css-nxb3y5 { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column;
background-color: red; }\`,
  ]),
};

const Vertical = props => <div className={\`\${styles.h1vjf5pn}\`} />;

export default Vertical;
"
`;

exports[`react-dom parses react-dom ViewAndListWithData 1`] = `
"import React from 'react';
import addressesData from './addresses.data.js';
import css from 'glam';
import userData from './user.data.js';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const ViewAndListWithData = props =>
  <div className={\`\${styles.h1lmw638}\`}>
    <div>{props.name}</div>
    <div className={\`\${styles.h1lmw638}\`}>
      {addressesData.map((item, i) => <div key={i}>{item.city}</div>)}
    </div>
  </div>;
ViewAndListWithData.defaultProps = userData;
export default ViewAndListWithData;
"
`;

exports[`react-dom parses react-dom ViewWithData 1`] = `
"import React from 'react';
import css from 'glam';
import userData from './user.data.js';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const ViewWithData = props =>
  <div className={\`\${styles.h1lmw638}\`}><div>{props.name}</div></div>;
ViewWithData.defaultProps = userData;
export default ViewWithData;
"
`;

exports[`react-dom parses react-dom When 1`] = `
"import React from 'react';

const When = props =>
  <div>{props.showMe ? <div>I may be hidden</div> : null}</div>;

export default When;
"
`;

exports[`react-dom parses react-dom WhenList 1`] = `
"import React from 'react';
import css from 'glam';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

const WhenList = props =>
  <div>
    {props.showMe
      ? <div className={\`\${styles.h1lmw638}\`}>
          {props.list.map((item, i) => <div key={i}>{item.thing}</div>)}
        </div>
      : null}
  </div>;

export default WhenList;
"
`;

exports[`react-dom parses react-dom WhenTopLevel 1`] = `
"import React from 'react';

const WhenTopLevel = props =>
  props.showMe ? <div>I may be hidden</div> : null;

export default WhenTopLevel;
"
`;

exports[`react-dom parses react-dom WithTest 1`] = `
"import React from 'react';
import css from 'glam';
import makeTests from './WithTest.view.tests.js';

const styles = {
  h1lmw638: css('css-1s0kfvh', [], () => [
    \`.css-1s0kfvh { flex-direction: column; -webkit-box-direction: normal; -webkit-box-orient: vertical; -ms-flex-direction: column; -webkit-flex-direction: column; }\`,
  ]),
};

class TestsWithTest extends React.Component {
  constructor(props) {
    super(props);

    this.tests = makeTests(this.display);
    this.state = this.tests[this.tests._main];
  }

  display = next => this.setState(next);

  render() {
    return <WithTest {...this.props} {...this.state} />;
  }
}

const WithTest = props =>
  <button onClick={props.change} className={\`\${styles.h1lmw638}\`}>
    <div>{props.text}</div>
  </button>;

export default TestsWithTest;
"
`;

exports[`react-native parses react-native Action 1`] = `
"import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const Action = props =>
  props.check
    ? <TouchableHighlight
        activeOpacity={0.7}
        onPress={props.clickMe}
        underlayColor=\\"transparent\\"
      >
        <View style={styles.h18pdll5}><Text>click me</Text></View>
      </TouchableHighlight>
    : null;

export default Action;
"
`;

exports[`react-native parses react-native BackgroundImage 1`] = `
"import React from 'react';
import { Image, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  h18pdll5: { flexDirection: 'column' },
  hp2cr4d: { flexDirection: 'row' },
});

const BackgroundImage = props =>
  <Image
    resizeMode=\\"cover\\"
    source={{ uri: 'https://viewsdx.com/image.jpg' }}
    style={styles.h18pdll5}
  >
    <Image
      resizeMode=\\"cover\\"
      source={{ uri: props.backgroundImage }}
      style={styles.hp2cr4d}
    />
  </Image>;

export default BackgroundImage;
"
`;

exports[`react-native parses react-native BoxShadow 1`] = `
"import React from 'react';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const BoxShadow = props =>
  <View
    style={[
      styles.h18pdll5,
      {
        elevation: 1,
        shadowOffset: { height: 10, width: 5 },
        shadowRadius: 2,
        shadowColor: props.color,
        shadowOpacity: 1,
      },
    ]}
  />;

export default BoxShadow;
"
`;

exports[`react-native parses react-native CaptureEmail 1`] = `
"import React from 'react';
import { KeyboardAvoidingView, StyleSheet, TextInput } from 'react-native';

const styles = StyleSheet.create({
  hm4pcs: {
    borderColor: 'black',
    borderStyle: 'solid',
    borderWidth: 1,
    color: 'red',
  },
});

const CaptureEmail = props =>
  <KeyboardAvoidingView behavior=\\"position\\">
    <TextInput
      onSubmitEditing={props.onSubmit}
      returnKeyType=\\"go\\"
      onChangeText={Email => this.setState({ Email })}
      ref={$e => (this.$captureEmail = $e)}
      value={state.Email}
      placeholderTextColor=\\"blue\\"
      style={styles.hm4pcs}
    />
  </KeyboardAvoidingView>;

export default CaptureEmail;
"
`;

exports[`react-native parses react-native CaptureFile 1`] = `
"import React from 'react';
import CaptureFile from './CaptureFile.view.js';

const CaptureFile = props => <CaptureFile />;

export default CaptureFile;
"
`;

exports[`react-native parses react-native CaptureNumber 1`] = `
"import React from 'react';
import { KeyboardAvoidingView, TextInput } from 'react-native';

const CaptureNumber = props =>
  <KeyboardAvoidingView behavior=\\"position\\">
    <TextInput />
  </KeyboardAvoidingView>;

export default CaptureNumber;
"
`;

exports[`react-native parses react-native CapturePhone 1`] = `
"import React from 'react';
import { KeyboardAvoidingView, TextInput } from 'react-native';

const CapturePhone = props =>
  <KeyboardAvoidingView behavior=\\"position\\">
    <TextInput />
  </KeyboardAvoidingView>;

export default CapturePhone;
"
`;

exports[`react-native parses react-native CaptureSecure 1`] = `
"import React from 'react';
import { KeyboardAvoidingView, TextInput } from 'react-native';

const CaptureSecure = props =>
  <KeyboardAvoidingView behavior=\\"position\\">
    <TextInput />
  </KeyboardAvoidingView>;

export default CaptureSecure;
"
`;

exports[`react-native parses react-native CaptureText 1`] = `
"import React from 'react';
import { KeyboardAvoidingView, TextInput } from 'react-native';

const CaptureText = props =>
  <KeyboardAvoidingView behavior=\\"position\\">
    <TextInput />
  </KeyboardAvoidingView>;

export default CaptureText;
"
`;

exports[`react-native parses react-native GoTo 1`] = `
"import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const GoTo = props =>
  <View style={styles.h18pdll5}><Text>Go to ViewsDX website!</Text></View>;

export default GoTo;
"
`;

exports[`react-native parses react-native Horizontal 1`] = `
"import React from 'react';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  h13qdyyc: {
    flexDirection: 'row',
    borderColor: 'red',
    borderStyle: 'solid',
    borderWidth: 1,
  },
});

const Horizontal = props => <View style={styles.h13qdyyc} />;

export default Horizontal;
"
`;

exports[`react-native parses react-native Hover 1`] = `
"import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';

const styles = StyleSheet.create({
  h1jrzrhl: { flexDirection: 'column', backgroundColor: 'black' },
  hyfo4ef: { color: 'white' },
});

const Hover = props =>
  <TouchableHighlight
    activeOpacity={0.7}
    onPress={props.onClick}
    underlayColor=\\"transparent\\"
  >
    <View style={styles.h1jrzrhl}>
      <Text style={styles.hyfo4ef}>Hey!</Text>
    </View>
  </TouchableHighlight>;

export default Hover;
"
`;

exports[`react-native parses react-native HoverProxy 1`] = `
"import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';

const styles = StyleSheet.create({
  h1jrzrhl: { flexDirection: 'column', backgroundColor: 'black' },
  hyfo4ef: { color: 'white' },
});

const HoverProxy = props =>
  <TouchableHighlight
    activeOpacity={0.7}
    onPress={props.onClick}
    underlayColor=\\"transparent\\"
  >
    <View style={styles.h1jrzrhl}>
      <props.icon from={props.icon} fill=\\"white\\" fillHover=\\"black\\" />
      <Text style={styles.hyfo4ef}>Hey!</Text>
    </View>
  </TouchableHighlight>;

export default HoverProxy;
"
`;

exports[`react-native parses react-native Image 1`] = `
"import React from 'react';
import { Image } from 'react-native';

const Image = props => <Image source=\\"https://viewsdx.com/rocks.jpg\\" />;

export default Image;
"
`;

exports[`react-native parses react-native JustText 1`] = `
"import React from 'react';
import { Text } from 'react-native';

const JustText = props => <Text>hey!</Text>;

export default JustText;
"
`;

exports[`react-native parses react-native ListOfSomething 1`] = `
"import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const ListOfSomething = props =>
  <View style={styles.h18pdll5}>
    {props.list.map((item, i) => <Text key={i}>{item.name}</Text>)}
  </View>;

export default ListOfSomething;
"
`;

exports[`react-native parses react-native ListWithData 1`] = `
"import React from 'react';
import usersData from './users.data.js';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const ListWithData = props =>
  <View style={styles.h18pdll5}>
    {usersData.map((item, i) => <Text key={i}>{item.name}</Text>)}
  </View>;

export default ListWithData;
"
`;

exports[`react-native parses react-native NameIsType 1`] = `
"import React from 'react';
import { Text } from 'react-native';

const NameIsType = props => <Text>iam</Text>;

export default NameIsType;
"
`;

exports[`react-native parses react-native NestedRoutes 1`] = `
"import React from 'react';
import { Link } from 'react-router-native';
import { Route } from 'react-router-native';
import Topic from './Topic.view.js';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const NestedRoutes = props =>
  <View>
    <Text>topics</Text>
    <Link activeOpacity={0.7} to=\\"rendering\\" underlayColor=\\"transparent\\">
      <View style={styles.h18pdll5}><Text>Rendering with React</Text></View>
    </Link>
    <Route
      path={\`\${props.match.url}/:topic\`}
      render={routeProps => <Topic {...routeProps} />}
    />
  </View>;

export default NestedRoutes;
"
`;

exports[`react-native parses react-native Print 1`] = `
"import React from 'react';
import { Text, View } from 'react-native';

const Print = props => <View><Text>not in print</Text><Text>hey</Text></View>;

export default Print;
"
`;

exports[`react-native parses react-native Proxy 1`] = `
"import React from 'react';
import { Text, View } from 'react-native';

const Proxy = props =>
  <View>
    <props.icon from={props.icon} fill=\\"red\\" /><Text>{props.text}</Text>
  </View>;

export default Proxy;
"
`;

exports[`react-native parses react-native Router 1`] = `
"import React from 'react';
import About from './About.view.js';
import Home from './Home.view.js';
import { Route } from 'react-router-native';
import { NativeRouter as Router } from 'react-router-native';
import Topics from './Topics.view.js';
import { View } from 'react-native';

const Router = props =>
  <Router>
    <View>
      <Route
        path=\\"/\\"
        exact
        render={routeProps =>
          <Home heigth={props.height} width={360} {...routeProps} />}
      />
      <Route path=\\"/about\\" render={routeProps => <About {...routeProps} />} />
      <Route path=\\"/topics\\" render={routeProps => <Topics {...routeProps} />} />
    </View>
  </Router>;

export default Router;
"
`;

exports[`react-native parses react-native Svg 1`] = `
"import React from 'react';
import Svg, { Circle, G, Path, Text as SvgText } from 'react-native-svg';

const Svg = props =>
  <Svg viewBox=\\"0 0 20 20\\">
    <Path d=\\"0 20\\" /><Circle cx={10} cy={5} r={5} />
    <G><Path d={1} /><Path d={2} /></G><SvgText text=\\"hey\\" />
  </Svg>;

export default Svg;
"
`;

exports[`react-native parses react-native Teleport 1`] = `
"import React from 'react';
import { Link } from 'react-router-native';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ hp2cr4d: { flexDirection: 'row' } });

const Teleport = props =>
  <Link activeOpacity={0.7} to=\\"..\\" underlayColor=\\"transparent\\">
    <View style={styles.hp2cr4d}><Text>go back</Text></View>
  </Link>;

export default Teleport;
"
`;

exports[`react-native parses react-native Text 1`] = `
"import React from 'react';
import { StyleSheet, Text } from 'react-native';

const styles = StyleSheet.create({
  h11t0jc: { color: 'blue', fontFamily: 'Montserrat-300', fontSize: 16 },
});

const Text = props => <Text style={styles.h11t0jc}>Hi, I'm some text</Text>;

export default Text;
"
`;

exports[`react-native parses react-native Toggle 1`] = `
"import React from 'react';
import { StyleSheet, TouchableHighlight } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

class RemapToggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isOn: props.isOn };
  }
  toggleIsOn = () => this.setState({ isOn: !this.state.isOn });

  render() {
    return (
      <Toggle {...this.props} {...this.state} toggleIsOn={this.toggleIsOn} />
    );
  }
}

const Toggle = props =>
  <TouchableHighlight
    activeOpacity={0.7}
    onPress={props.toggleIsOn}
    underlayColor=\\"transparent\\"
  >
    <undefined style={styles.h18pdll5} />
  </TouchableHighlight>;

export default RemapToggle;
"
`;

exports[`react-native parses react-native UseOfProxy 1`] = `
"import React from 'react';
import Button from './Button.view.js';
import { View } from 'react-native';

const UseOfProxy = props =>
  <View><Button color=\\"red\\" icon=\\"MyIcon\\" text=\\"I'm a button!\\" /></View>;

export default UseOfProxy;
"
`;

exports[`react-native parses react-native Vertical 1`] = `
"import React from 'react';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  h1c5r919: { flexDirection: 'column', backgroundColor: 'red' },
});

const Vertical = props => <View style={styles.h1c5r919} />;

export default Vertical;
"
`;

exports[`react-native parses react-native ViewAndListWithData 1`] = `
"import React from 'react';
import addressesData from './addresses.data.js';
import userData from './user.data.js';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const ViewAndListWithData = props =>
  <View style={styles.h18pdll5}>
    <Text>{props.name}</Text>
    <View style={styles.h18pdll5}>
      {addressesData.map((item, i) => <Text key={i}>{item.city}</Text>)}
    </View>
  </View>;
ViewAndListWithData.defaultProps = userData;
export default ViewAndListWithData;
"
`;

exports[`react-native parses react-native ViewWithData 1`] = `
"import React from 'react';
import userData from './user.data.js';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const ViewWithData = props =>
  <View style={styles.h18pdll5}><Text>{props.name}</Text></View>;
ViewWithData.defaultProps = userData;
export default ViewWithData;
"
`;

exports[`react-native parses react-native When 1`] = `
"import React from 'react';
import { Text, View } from 'react-native';

const When = props =>
  <View>{props.showMe ? <Text>I may be hidden</Text> : null}</View>;

export default When;
"
`;

exports[`react-native parses react-native WhenList 1`] = `
"import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

const WhenList = props =>
  <View>
    {props.showMe
      ? <View style={styles.h18pdll5}>
          {props.list.map((item, i) => <Text key={i}>{item.thing}</Text>)}
        </View>
      : null}
  </View>;

export default WhenList;
"
`;

exports[`react-native parses react-native WhenTopLevel 1`] = `
"import React from 'react';
import { Text } from 'react-native';

const WhenTopLevel = props =>
  props.showMe ? <Text>I may be hidden</Text> : null;

export default WhenTopLevel;
"
`;

exports[`react-native parses react-native WithTest 1`] = `
"import React from 'react';
import { StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import makeTests from './WithTest.view.tests.js';

const styles = StyleSheet.create({ h18pdll5: { flexDirection: 'column' } });

class TestsWithTest extends React.Component {
  constructor(props) {
    super(props);

    this.tests = makeTests(this.display);
    this.state = this.tests[this.tests._main];
  }

  display = next => this.setState(next);

  render() {
    return <WithTest {...this.props} {...this.state} />;
  }
}

const WithTest = props =>
  <TouchableHighlight
    activeOpacity={0.7}
    onPress={props.change}
    underlayColor=\\"transparent\\"
  >
    <View style={styles.h18pdll5}><Text>{props.text}</Text></View>
  </TouchableHighlight>;

export default TestsWithTest;
"
`;
